src/app/public/modules/data-manager/models/data-manager-state-change.ts

```diff
@@ -0,0 +1,13 @@
+import {
+  SkyDataManagerState
+} from '.';
+
+export class SkyDataManagerStateChange {
+  public dataState: SkyDataManagerState;
+  public source: string;
```

Comment:
How will `source` be used? Should this string have some predefined values?

Reply:
`source` is basically just a mechanism to keep views/components from responding to their own changes. If the toolbar made a change, then saw an update from the service (that it caused) it would at the least cause unnecessary changes, at the worst in some situations creates an infinite loop. We could have some predefined values for the components we control (toolbar, data manager) but need to keep it open-ended so consumers can set a source for their views. Would `sourceId` make it more clear?

Reply:
That makes sense. Thanks for the explanation. I think `source` is fine for now.

Reply:
Since there is a lot under the hood on this one, can we make sure this property is documented well?

Reply:
the param is well explained in the consumer-facing docs, did you mean leaving a comment in the actual documentation?

Reply:
As long as its spelled out somewhere, that's great. Thanks!